src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -14,13 +14,21 @@ public class StringFunctions {
 
     public static final String reverseWithStringConcat(String string) {
         String output = new String();
+       
+        if (string == null || string.isEmpty()) { //If the string is null, return an empty string
+            return "";
+        }
+       
         for (int i = (string.length() - 1); i >= 0; i--) {
             output += (string.charAt(i));
         }
         return output;
     }
 
     public static final String reverseWithStringBuilder(String string) {
+        if (string == null || string.isEmpty()) { //If the string is null, return an empty string
```

Comment:
This string uses the same if statement used in the reverseWithStringConcat function to check if the string is null, if it is return an empty string